connmgr/connmanager_test.go

```diff
@@ -133,18 +151,64 @@ func TestStartStop(t *testing.T) {
 	}
 }
 
+func overrideActiveConfig() func() {
+	originalActiveCfg := config.ActiveConfig()
+	config.SetActiveConfig(&config.Config{
+		Flags: &config.Flags{
+			NetworkFlags: config.NetworkFlags{
+				ActiveNetParams: &dagconfig.SimnetParams},
+		},
+	})
+	return func() {
+		// Give some extra time to all open NewConnReq goroutines
+		// to finish before restoring the active config to prevent
+		// potential panics.
+		time.Sleep(10 * time.Millisecond)
+
+		config.SetActiveConfig(originalActiveCfg)
+	}
+}
+
+func addressManagerForTest(t *testing.T, testName string, numAddresses uint8) *addrmgr.AddrManager {
+	amgr := emptyAddressManagerForTest(t, testName)
+
+	for i := uint8(0); i < numAddresses; i++ {
+		ip := fmt.Sprintf("173.%d.115.66:16511", i)
+		err := amgr.AddAddressByIP(ip, nil)
+		if err != nil {
+			t.Fatalf("AddAddressByIP unexpectedly failed to add IP %s: %s", ip, err)
+		}
+	}
+
+	return amgr
+}
+
+func emptyAddressManagerForTest(t *testing.T, testName string) *addrmgr.AddrManager {
```

Comment:
Empty is a verb. Please rename it to something like "createAddressManagerForTest"